### PR TITLE
Ban zero blob transactions

### LINKED ISSUE #5401: Ban Zero Blob Transactions
### Description

Zero blob transaction are not valid, see
https://github.com/ethereum/EIPs/pull/6863

### PR DESCRIPTION
**Pull request by fab-10**
## PR description

[An update to EIP-4844](https://github.com/ethereum/EIPs/pull/6863) made zero blob transactions invalid, so this PR adds the check to verify that at least one blob is specified, this check could be done statically when creating the transaction.

## Fixed Issue(s)

fixes #5401

### CODE REVIEW

File: ethereum/core/src/test-support/java/org/hyperledger/besu/ethereum/core/TransactionTestFixture.java [hunk 1]
```diff
@@ -53,18 +61,35 @@ public Transaction createTransaction(final KeyPair keys) {
...
+        builder.maxPriorityFeePerGas(maxPriorityFeePerGas.orElse(Wei.of(500)));
+        builder.maxFeePerGas(maxFeePerGas.orElse(Wei.of(5000)));
+        builder.accessList(accessListEntries.orElse(List.of()));
+        break;
+      case BLOB:
+        builder.maxPriorityFeePerGas(maxPriorityFeePerGas.orElse(Wei.of(500)));
+        builder.maxFeePerGas(maxFeePerGas.orElse(Wei.of(5000)));
+        builder.accessList(accessListEntries.orElse(List.of()));
+        builder.maxFeePerDataGas(maxFeePerDataGas.orElse(Wei.ONE));
+        builder.versionedHashes(versionedHashes.orElse(List.of(Hash.wrap(Bytes32.random()))));
```
Comment (shemnon) [inline ethereum/core/src/test-support/java/org/hyperledger/besu/ethereum/core/TransactionTestFixture.java:87]: Random will produce variable test data.  I'd recommend something else like a converted ascii string "Versioned Hash Not Set", padded to 32 bytes.

Also, when random the version byte will almost always be invalid.

**Comment by shemnon** [comment]
Unit tests need help.  `P2PPlainNetworkTest` and `TLSContextFactoryTest ` appear to be because of expired test certs.  But `TransactionCompleteResultTest` feels like a quality hit.

File: ethereum/core/src/test-support/java/org/hyperledger/besu/ethereum/core/TransactionTestFixture.java (hunk 1 above)
Comment (fab-10) [reply ethereum/core/src/test-support/java/org/hyperledger/besu/ethereum/core/TransactionTestFixture.java:87]: Right, I forgot to properly format the versioned hash, fixing

**Comment by fab-10** [comment]
> Unit tests need help. `P2PPlainNetworkTest` and `TLSContextFactoryTest` appear to be because of expired test certs. But `TransactionCompleteResultTest` feels like a quality hit.

Test fixed

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java [hunk 2]
```diff
@@ -504,6 +504,29 @@ public void shouldRejectTooLargeInitcode() {
...
+        new MainnetTransactionValidator(
+            gasCalculator,
+            GasLimitCalculator.constant(),
+            FeeMarket.london(0L),
+            false,
+            Optional.of(BigInteger.ONE),
+            Set.of(TransactionType.FRONTIER, TransactionType.EIP1559, TransactionType.BLOB),
+            0xc000);
+
+    var zeroBlobTx =
```
Comment (macfarla) [inline ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java:519]:
```suggestion
    var blobTx =
```

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java [hunk 3]
```diff
@@ -504,6 +504,29 @@ public void shouldRejectTooLargeInitcode() {
...
+        new TransactionTestFixture()
+            .type(TransactionType.BLOB)
+            .chainId(Optional.of(BigInteger.ONE))
+            .createTransaction(senderKeys);
+    var validationResult =
+        validator.validate(zeroBlobTx, Optional.empty(), transactionValidationParams);
```
Comment (macfarla) [inline ethereum/core/src/test/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidatorTest.java:525]:
```suggestion
        validator.validate(blobTx, Optional.empty(), transactionValidationParams);
```

**Review by macfarla — APPROVED** [review]
LGTM. one comment on a var name in test

**Pull request merged**